quickwit/quickwit-index-management/src/garbage_collection.rs

```diff
@@ -219,9 +229,32 @@ async fn delete_splits(
     while let Some(delete_split_result) = delete_split_from_index_res_stream.next().await {
         match delete_split_result {
             Ok(entries) => {
+                if let Some(metrics) = metrics {
```

Comment:
can we move this logic as a method of GcMetrics?

Something like 

`fn record_successful_deletion(&mut self, num_delete_splits: usize, num_deleted_bytes: u64);`